### PR TITLE
Move ParseLink and validators into runconfig.parse where they are used

### PR DESCRIPTION
**Pull request by dnephin**
This change (with #18693) removes the transitive dependency from `api/client` on `pkg/parsers`.

`parsers.ParseLink` and `opts.ValidateLink` were moved to break the dependency.

`opts.ValidDeviceMode`, and `opts.ValidateDevice`  were moved to the single place they were called (`runconfig`) to reduce the scope of `opts`. They could be made private now.

cc @calavera, @tiborvass

### CODE REVIEW

File: daemon/daemon_unix.go [hunk 1]
```diff
@@ -648,7 +647,7 @@ func (daemon *Daemon) registerLinks(container *container.Container, hostConfig *
 	}
 
 	for _, l := range hostConfig.Links {
-		name, alias, err := parsers.ParseLink(l)
+		name, alias, err := runconfig.ParseLink(l)
```
Comment (tiborvass) [inline daemon/daemon_unix.go]: @calavera are you fine with moving ParseLink to runconfig?

Comment (calavera) [inline daemon/daemon_unix.go]: yep, no problem.

**Comment by calavera** [comment]
LGTM

**Comment by vdemeester** [comment]
@dnelphin needs a rebase :sweat:

**Comment by dnephin** [comment]
Ok, rebased, and the build is green (except for windows, which seemed to fail on setup, not during a test run)

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**